demo/test.c

```diff
@@ -2294,37 +2294,112 @@ static int test_mp_radix_size(void)
    /* number to result in a different size for every base: 67^(4 * 67) */
    mp_set(&a, 67);
    if ((err = mp_expt_u32(&a, 268u, &a)) != MP_OKAY) {
-      goto LBL_ERR;
+      goto LTM_ERR;
    }
 
    for (radix = 2; radix < 65; radix++) {
       if ((err = mp_radix_size(&a, radix, &size)) != MP_OKAY) {
-         goto LBL_ERR;
+         goto LTM_ERR;
       }
       if (size != results[radix]) {
          fprintf(stderr, "mp_radix_size: result for base %d was %d instead of %d\n",
                  radix, size, results[radix]);
-         goto LBL_ERR;
+         goto LTM_ERR;
       }
       a.sign = MP_NEG;
       if ((err = mp_radix_size(&a, radix, &size)) != MP_OKAY) {
-         goto LBL_ERR;
+         goto LTM_ERR;
       }
       if (size != (results[radix] + 1)) {
          fprintf(stderr, "mp_radix_size: result for base %d was %d instead of %d\n",
                  radix, size, results[radix]);
-         goto LBL_ERR;
+         goto LTM_ERR;
       }
       a.sign = MP_ZPOS;
    }
 
    mp_clear(&a);
    return EXIT_SUCCESS;
-LBL_ERR:
+LTM_ERR:
    mp_clear(&a);
    return EXIT_FAILURE;
 }
 
+#ifndef S_MP_DIV_SMALL
+/* Some larger values to test the fast division algorithm */
+static int test_s_mp_div(void)
+{
+   mp_int a, b, c, d;
+   int size, err;
+
+   if ((err = mp_init_multi(&a, &b, &c, &d, NULL)) != MP_OKAY) {
+      goto LTM_ERR;
+   }
+
+   for (size = MP_KARATSUBA_MUL_CUTOFF; size < 3 * MP_KARATSUBA_MUL_CUTOFF; size += 10) {
+      fprintf(stderr,"sizes = %d / %d\n", 10 * size, size);
+      /* Relation 10:1 */
+      if ((err = mp_rand(&a, 10 * size)) != MP_OKAY) {
+         goto LTM_ERR;
+      }
+      if ((err = mp_rand(&b, size)) != MP_OKAY) {
+         goto LTM_ERR;
+      }
+      if ((err = s_mp_div_recursive(&a, &b, &c, NULL)) != MP_OKAY) {
+         goto LTM_ERR;
+      }
+      if ((err = s_mp_div_school(&a, &b, &d, NULL)) != MP_OKAY) {
+         goto LTM_ERR;
+      }
+      if (mp_cmp(&c, &d) != MP_EQ) {
+         fprintf(stderr, "1. Recursive division failed at sizes %d / %d\n", 10 * size, size);
+         goto LTM_ERR;
+      }
+      fprintf(stderr,"sizes = %d / %d\n", 2 * size, size);
+      /* Relation 2:1 */
+      if ((err = mp_rand(&a, 2 * size)) != MP_OKAY) {
+         goto LTM_ERR;
+      }
+      if ((err = mp_rand(&b, size)) != MP_OKAY) {
+         goto LTM_ERR;
+      }
+      if ((err = s_mp_div_recursive(&a, &b, &c, NULL)) != MP_OKAY) {
+         goto LTM_ERR;
```

Comment:
Compare the mod too? Maybe write this function as a helper with function pointers and then call div_test(school, recursive)
div_test(school, small)?

Reply:
> Compare the mod too?

Yepp, can do, no prob'.

> Maybe write this function as a helper with function pointers and then call div_test(school, recursive), div_test(school, small)?

Well, C&P with replace would be simpler, wouldn't it? ;-)
`div_small` is quite slow and as it does not get used with recursion (save a couple of bytes with `div_small` and than add a load full with recursion?) it might need too much runtime but haven't checked yet.